scripts/test_db_connection_pool_load.sh

```
@@ -98,16 +98,40 @@ monitor_resources() {
-        local cpu=$(top -bn1 | grep "Cpu(s)" | sed "s/.*, *\([0-9.]*\)%* id.*/\1/" | awk '{print 100 - $1}')
+        # CPU usage - portable detection
+        local cpu
+        if [[ "$OSTYPE" == "darwin"* ]]; then
+            # macOS
+            cpu=$(top -l 1 | grep "CPU usage" | awk '{print $3}' | sed 's/%//')
+        elif command -v mpstat >/dev/null 2>&1; then
+            # Linux with mpstat
+            cpu=$(mpstat 1 1 | awk '/Average/ {print 100 - $NF}')
+        else
+            # Fallback to /proc/stat
+            local stat1=($(cat /proc/stat | head -1))
```

Shellcheck (0.10.0)

[warning] Prefer mapfile or read -a to split command output (or quote to avoid splitting).

(SC2207)